src/plugins/intel_gpu/src/kernel_selector/jitter.cpp

```diff
@@ -578,8 +578,8 @@ JitDefinitions DataTensorJitConstant::GetDefinitions() const {
         } else {
             auto f_pad = toCodeString(_tensor.Feature().pad.before);
             auto f_size = toCodeString(_tensor.Feature().v);
-            definitions.push_back({ safe_index_func_name, "((" + f_pad + " + (f)) % " + f_size + ")" });
-            definitions.push_back({ index_func_name, "(" + toCodeString(_tensor.Feature().pad.before) + " + (f))" });
+            definitions.push_back({ safe_index_func_name, "(" + offset + " + ((" + f_pad + " + (f)) % " + f_size + "))" });
+            definitions.push_back({ index_func_name, "(" + offset + " + " + toCodeString(_tensor.Feature().pad.before) + " + (f))" });
```

Comment:
If I understood correctly, `offset` already includes `f_pad`.

Reply:
Seems like you're right, but as I can see in case when f_pad != 0 another code branch is used. Anyway, fixed, thanks!